src/main/java/org/eln2/mc/common/containers/ContainerRegistry.kt

```diff
@@ -0,0 +1,87 @@
+package org.eln2.mc.common.containers
+
+import net.minecraft.client.gui.screens.MenuScreens
+import net.minecraft.client.gui.screens.inventory.AbstractContainerScreen
+import net.minecraft.network.chat.Component
+import net.minecraft.world.entity.player.Inventory
+import net.minecraft.world.inventory.AbstractContainerMenu
+import net.minecraft.world.inventory.MenuType
+import net.minecraftforge.eventbus.api.IEventBus
+import net.minecraftforge.eventbus.api.SubscribeEvent
+import net.minecraftforge.fml.event.lifecycle.FMLClientSetupEvent
+import net.minecraftforge.registries.DeferredRegister
+import net.minecraftforge.registries.ForgeRegistries
+import net.minecraftforge.registries.RegistryObject
+import org.eln2.mc.Eln2
+import org.eln2.mc.client.screens.CapacitorCellScreen
+import org.eln2.mc.client.screens.InductorCellScreen
+import org.eln2.mc.client.screens.ResistorCellScreen
+import org.eln2.mc.client.screens.VoltageSourceCellScreen
+import org.eln2.mc.common.blocks.BlockRegistry
+
+object ContainerRegistry {
+    @JvmStatic
+    val CONTAINER_REGISTRY: DeferredRegister<MenuType<*>> = DeferredRegister.create(
+        ForgeRegistries.CONTAINERS,
+        Eln2.MODID
+    )
+    private val CLIENTSIDE_GUI_QUEUE = mutableListOf<ContainerRegistryItem<*>>()
+
+    fun setup(bus: IEventBus) = CONTAINER_REGISTRY.register(bus)
+
+    data class ContainerRegistryItem<T: AbstractContainerMenu>(
+        val name: String,
+        val screenConstructor: (T, Inventory, Component) -> AbstractContainerScreen<T>,
+        val container: RegistryObject<MenuType<T>>
+    )
```

Comment:
I've been moving these out of the parent class because it helps discoverability of files in the project file tree and makes individual files easier to read (since it's somewhat shorter). They are also kinda an API-like thing so I'd like to comment them, but nothing we need to do before a refactor.

Reply:
This has been removed. The new register method only really exists to avoid having to explicitly type each registry object